rest-client/src/main/java/io/apicurio/registry/client/RegistryRestClientImpl.java

```diff
@@ -231,133 +237,136 @@ public ArtifactMetaData createArtifact(String artifactId, ArtifactType artifactT
     @Override
     public ArtifactMetaData createArtifact(String artifactId, ArtifactType artifactType, InputStream data,
                                            IfExistsType ifExists, Boolean canonical) {
+        if (artifactId != null && !ArtifactIdValidator.isArtifactIdAllowed(artifactId)) {
+            throw new InvalidArtifactIdException();
+        }
```

Comment:
Check it in the client to fail-fast, smart.

Reply:
not so smart, non-ASCII chars are forbidden because of okhttp which don't allow those chars in http headers